lib/tests/streamlit/caching_test.py

```diff
@@ -39,6 +39,28 @@ def foo():
         self.assertEqual(foo(), 42)
         self.assertEqual(foo(), 42)
 
+    def test_implicit_inputs(self):
```

Comment:
Maybe we should break this into two separate tests so it's a bit easier to grok.

Also, should we update or mimic the following tests in `hashing_test.py`?
```
test_referenced
test_referenced_referenced
test_lambdas
test_dict_reference
```